docs/registry/create/_index.md

```diff
@@ -11,154 +11,110 @@ tags:
-
-See [Valid APIs to implement in your model](#valid-apis-to-implement-in-your-model) for more information.
-
-{{% /tab %}}
-{{% tab name="C++" %}}
-
-To create a new resource model, start by referencing the corresponding built-in resource's implementation in the [Viam C++ SDK](https://github.com/viamrobotics/viam-cpp-sdk/tree/main/src/viam/sdk).
-
-For example, if you are implementing a new custom `base` component model, reference the [<file>components/base/base.hpp</file>](https://github.com/viamrobotics/viam-cpp-sdk/blob/main/src/viam/sdk/components/base/base.hpp) implementation file to learn how the `BaseClient` class is implemented.
-
-You can create your own derivative class in a <file>.cpp</file>, such as <file>my_modular_resource.cpp</file>, and optionally define the implementation of that class in a <file>.hpp</file> header file, such as <file>my_modular_resource.hpp</file>.
-The example code in the following sections demonstrates an implementation using both files.
-
-See [Valid APIs to implement in your model](#valid-apis-to-implement-in-your-model) for more information.
-
-{{% /tab %}}
-{{< /tabs >}}
-
-1. **Prepare to Import Your Custom Model and Subtype's API:**
-
-   To prepare to import your custom model and chosen resource subtype's API into your main program and register them with your preferred SDK:
-
-   {{< tabs >}}
-   {{% tab name="Python" %}}
-
-Find the subtype API as defined in the relevant <file>components/\<resource-name\>/\<resource-name>\.py</file> or <file>services/\<resource-name\>/<resource-name>.py</file> file in the [Python SDK](https://github.com/viamrobotics/viam-python-sdk).
-
-For example, the `base` component API is defined in [<file>components/base/base.py</file>](https://github.com/viamrobotics/viam-python-sdk/blob/main/src/viam/components/base/base.py#L11).
-
-{{% /tab %}}
-{{% tab name="Go" %}}
-
-Find the subtype API as defined in the relevant <file>components/\<resource-name\>/\<resource-name\>.go</file> or <file>services/\<resource-name\>/\<resource-name\>.go</file> file in the [RDK](https://github.com/viamrobotics/rdk).
-
-For example, the `base` component API is defined in [<file>components/base/base.go</file>](https://github.com/viamrobotics/rdk/blob/main/components/base/base.go#L37).
-
-{{% /tab %}}
-{{% tab name="C++" %}}
-
-Find the subtype API as defined in the relevant <file>components/\<resource-name\>/\<resource-name>\.hpp</file> or <file>services/\<resource-name\>/<resource-name>.hpp</file> implementation (header) file in the [C++ SDK](https://github.com/viamrobotics/viam-cpp-sdk/).
+{{% alert title=Note color="note" %}}
+If you want to write a module to extend support to a new type of component that is relatively unique, and doesn't readily correspond to an existing [built-in component API](/build/program/apis/#component-apis), you may want to consider using the [generic API](/components/generic/) with your module instead of extending an existing API.
```

Comment:
```suggestion
If you want to write a module to extend support to a new type of component that is relatively unique, and doesn't readily correspond to an existing [built-in component API](/build/program/apis/#component-apis), consider using the [generic API](/components/generic/) with your module instead of extending an existing API.
```